dms/models/directory.py

```diff
@@ -548,7 +555,9 @@ def _check_directory_access(self):
     @api.constrains("name")
     def _check_name(self):
         for record in self:
-            if not check_name(record.name):
+            if not self.env.context.get("bypass_check_name") and not check_name(
```

Comment:
[I'm generally against negative booleans](https://www.recallstack.icu/en/2020/09/01/do-not-use-negative-booleans-please/). Better:

```suggestion
            if self.env.context.get("check_name", True) and not check_name(
```